content/2023-12-28-embedded-hal-v1.md

```diff
@@ -0,0 +1,77 @@
+companion crates `embedded-hal-bus`, `embedded-hal-async` and `embedded-hal-nb`.
+
+Check out the [repository](https://github.com/rust-embedded/embedded-hal), the [API documentation](https://github.com/rust-embedded/embedded-hal#crates) and the [migration guide](https://github.com/rust-embedded/embedded-hal/blob/master/docs/migrating-from-0.2-to-1.0.md).
+
+The `embedded-hal` crates provide traits (interfaces) for using peripherals commonly available in microcontrollers
+such as GPIO, UART, SPI or I2C. They allow writing drivers (for sensors, displays, actuators, network adapters, etc.) in
+a generic way, so they work on any microcontroller with an `embedded-hal` implementation without
+modifying them. It's a central piece of the Embedded Rust ecosystem, ensuring interoperability throughout.
+
+The 1.0 release has been in the works since 2020. Now that it's out, we consider all
+traits in it to be *stable*. The plan is to extend `embedded-hal` with more traits in future 1.x releases,
+not doing more breaking changes (i.e. there are no plans for a a 2.0 release). This will provide a stable
+base for building HALs and drivers.
+
+So, what's new in `embedded-hal` 1.0?
+
+<!-- more -->
+
+## Focus on drivers
+
+Previous versions of `embedded-hal` had a dual goal of standardizing HAL APIs for end users, and allowing writing generic drivers. 
+Experience has shown that these goals sometimes conflict with each other. As the latter brings much more value, 1.0 focuses on that.
+
+We've simplified some traits and merged others to remove interoperability gotchas.
+
+We've removed traits that were found to not be usable for generic drivers (most notably timers). The plan is to add
+them back in the future, with a better design. See the [migration guide](https://github.com/rust-embedded/embedded-hal/blob/master/docs/migrating-from-0.2-to-1.0.md) for details and links to the tracking issues.
+
+## Async
+
+A new addition is the `embedded-hal-async` crate, containing async versions of the traits. With today's Rust 1.75 release,
+async traits are available on Rust stable. They can be used without heap allocations or dynamic
+dispatch (unlike previous macro-based polyfills like the `async-trait` crate), so they are a great fit for bare-metal embedded usage.
+
+Most `embedded-hal-async` traits are async versions of their blocking counterparts.
+
+However, one highlight is the [`digital::Wait`](https://docs.rs/embedded-hal-async/1.0.0/embedded_hal_async/digital/trait.Wait.html) trait, with methods like `wait_for_high()` and `wait_for_low()`. This trait adds support for using "IRQ" GPIO pins typically used by SPI and I2C devices to send an interrupt to the microcontroller. This frequently-requested feature turned out to be hard to abstract with traits, but is now feasible in an ergonomic way with async.
+
+## SPI bus sharing
+
+![SPI bus/device diagram](../spi-shared-bus.svg)
```

Comment:
Hm, is that path okay? Currently the image would end up on the same level.

Reply:
zola makes the url end up like `http://127.0.0.1:1111/embedded-hal-v1/` so the `../` is needed, I've checked it works with `zola serve`.